src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface.jl

```diff
@@ -245,9 +239,35 @@ struct ForwardBackwardScheme end
 end
 
 @inline cosine_averaging_kernel(τ::FT) where FT = τ >= 0.5 && τ <= 1.5 ? FT(1 + cos(2π * (τ - 1))) : zero(FT)
-
 @inline constant_averaging_kernel(τ) = 1
 
+struct AdaptiveSubsteps{B, F}
+    Δtᴮ :: B
+    averaging_kernel :: F
+end
```

Comment:
Do we also need something like `minimum_substeps`?

Reply:
we can probably hardcode that.